docs/api-reference/next/router.md

```diff
@@ -154,6 +154,55 @@ Router.replace('/home')
 
 The API for `Router.replace` is exactly the same as that used for [`Router.push`](#router.push).
 
+### Router.prefetch
+
+Prefetch pages for faster client-side transitions. This method is only useful for navigations without [`next/link`](/docs/api-reference/next/link.md), as the later takes care of prefetching pages for you.
+
+> this is a production only feature, Next.js doesn't prefetch pages on development.
+
+```jsx
+import Router from 'next/router'
+
+Router.prefetch(url, as)
+```
+
+- `url` - The path to a `page`. that is, the name of a page inside the `pages` directory
+- `as` - Optional decorator for `url`, used to prefetch [dynamic routes](/docs/routing/dynamic-routes). Defaults to `url`
+
+#### Usage
+
+Let's say you have a login page, and after a login you redirect the user to the dashboard, for that case, we can prefetch the dashboard to make a faster transition, like in the following example:
+
+```jsx
+import Router from 'next/router'
+
+export default function Login() {
+  const handleSubmit = e => {
+    e.preventDefault()
+
+    fetch('/api/login', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        /* Form data */
+      }),
+    }).then(res => {
+      // Do a fast client-side transition to the already prefetched dashbaord page
+      if (res.ok) Router.push('/dashboard')
+    })
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      {/* Form fields */}
+      <button type="submit">Login</button>
+      {// Prefetch the dashboard page as the user will go there after the login
+      Router.prefetch('/dashboard')}
```

Comment:
Should we really put `Router.prefetch('...')` here (as part of the rendered code)? Why not use `useEffect()` as it's a side effect?